pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java

```diff
@@ -124,22 +124,49 @@ public void setLogWriter(PrintWriter printWriter) {
    * Gets the name of the host the PostgreSQL database is running on.
    *
    * @return name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #getServerNames()}
    */
+  @Deprecated
   public String getServerName() {
-    return serverName;
+    return serverNames[0];
+  }
+
+  /**
+   * Gets the name of the host(s) the PostgreSQL database is running on.
+   *
+   * @return name of the host(s) the PostgreSQL database is running on
+   */
+  public String[] getServerNames() {
+    return serverNames;
   }
 
   /**
    * Sets the name of the host the PostgreSQL database is running on. If this is changed, it will
    * only affect future calls to getConnection. The default value is <tt>localhost</tt>.
    *
    * @param serverName name of the host the PostgreSQL database is running on
+   * @deprecated use {@link #setServerNames(String[])}
    */
+  @Deprecated
   public void setServerName(String serverName) {
     if (serverName == null || serverName.equals("")) {
-      this.serverName = "localhost";
+      this.serverNames = new String[] {"localhost"};
     } else {
-      this.serverName = serverName;
+      this.serverNames = new String[] {serverName};
+    }
```

Comment:
Why allocate a new array? You already have created the array; why not just set the array element ?

Reply:
to make that exactly right, I would have to check if the array length is really exactly 1, then I can just set the element, right; otherwise I need to create the array anew anyway.
do you think the additional check is preferable ?

Reply:
Nah, leave it like it is, I should have read the entire thing before commenting